src/NodeServices/PythonServices.cs

```diff
@@ -164,73 +171,39 @@ private PythonEngineManager()
         {
             AvailableEngines = new ObservableCollection<PythonEngine>();
 
-            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
-            {
-                ScanPythonEngine(assembly);
-            }
-            AppDomain.CurrentDomain.AssemblyLoad += new AssemblyLoadEventHandler(AssemblyLoadEventHandler);
-        }
+            LoadPythonEngine(AppDomain.CurrentDomain.GetAssemblies().
+                FirstOrDefault(a => a != null && a.GetName().Name == CPythonAssemblyName));
 
-        private PythonEngine GetEngine(PythonEngineVersion version)
-        {
-            return GetEngine(version.ToString());
+            AppDomain.CurrentDomain.AssemblyLoad += new AssemblyLoadEventHandler((object sender, AssemblyLoadEventArgs args) =>
+            {
+                if (args.LoadedAssembly != null && 
+                    args.LoadedAssembly.GetName().Name == CPythonAssemblyName)
+                {
+                    Instance.LoadPythonEngine(args.LoadedAssembly);
+                }
+            });
```

Comment:
Why are you only checking for CPython? What if other assemblies are being loaded that contain other python evaluators?

Reply:
We only check for the Default python engine because that is the only one loaded through static references.
If we were to change it so that CPythonEnigne would be a Built in package or an extension...we could get rid of this